src/mudlet.pro

```diff
@@ -71,8 +71,12 @@ macx {
     QMAKE_CFLAGS_DEBUG += -O0
 }
 
-# enable C++14 for builds.
-CONFIG += c++14
+# enable C++17 for builds.
+lessThan(QT_MAJOR_VERSION, 5)|if(lessThan(QT_MAJOR_VERSION,6):lessThan(QT_MINOR_VERSION, 12)) {
```

Comment:
Is that "every QT version < 6.12"? Or am I misreading something here?

Reply:
It's confusing to me too, but that's the check we have for detecting if we're below Qt 5.7 and it works.

Reply:
Ah right... It's not `< 6.12` but effectively `major = 5 && minor < 12`

Reply:
Right. Poor code, but the best we can do.

Reply:
As I read it it seems to say, IF the Qt Major version is less than **5** - so if for *any* Qt 4.x libraries OR IF the Qt Major version is less than **6** AND the Qt Minor version is less than **12** use the first scope which is:
```
    QMAKE_CXXFLAGS += -std=c++17
```
i.e. bodge the CXXFLAGS for the compiler (or rather any compiler that considers that value so: `GCC`, `Clang` OR the `mingw-gcc` one) to that.

However, if the Qt Version is **5.12** or later, use the `CONFIG += c++17` option instead which `qmake` will process as required to make the settings right to use the C++17 standard.

What this is suggesting is that it seems that prior to 5.12 Qt did not handle C++17 directly within `qmake`.

Reply:
Yep indeed.